ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,98 +6,134 @@ set -o pipefail
-| .spec.template.metadata.labels.\"machine.openshift.io/cluster-api-machineset\" = .metadata.name" \
-${dir}/99_openshift-cluster-api_worker-machineset-0.yaml > ${dir}/99_openshift-cluster-api_worker-machineset-additional.yaml
+case $CLUSTER_TYPE in
+*aws*)
+  echo "Extracting AMI..."
+  REGION=${LEASED_RESOURCE}
+  amiid_workers_additional=$(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -oyaml | \
+    yq-v4 ".data.stream
+      | eval(.).architectures.${ADDITIONAL_WORKER_ARCHITECTURE}.images.aws.regions.\"${REGION}\".image")
 
-yq-v4 eval ".spec.template.spec.providerSpec.value.ami.id = \"${amiid_workers_additional}\"
-            | .spec.template.spec.providerSpec.value.instanceType = \"${ADDITIONAL_WORKER_VM_TYPE}\"
-            " -i ${dir}/99_openshift-cluster-api_worker-machineset-additional.yaml
+  echo "Updating the machineset with ${ADDITIONAL_WORKER_VM_TYPE} and ami ${amiid_workers_additional} ..."
 
-echo "Creating ${ADDITIONAL_WORKER_ARCHITECTURE} worker MachineSet"
-exec oc create -f ${dir}/99_openshift-cluster-api_worker-machineset-additional.yaml &
+  MACHINE_SET=$(yq-v4 ".spec.template.spec.providerSpec.value.ami.id = \"${amiid_workers_additional}\"
+                     | .spec.template.spec.providerSpec.value.instanceType = \"${ADDITIONAL_WORKER_VM_TYPE}\"
+              " <<< "${MACHINE_SET}")
+;;
+*azure*)
+  echo "az version:"
+  az version
+  azure_auth_location=$CLUSTER_PROFILE_DIR/osServicePrincipal.json
+  echo "Logging in with az"
+  azure_auth_client_id=$(yq-v4 .clientId < "$azure_auth_location")
+  azure_auth_client_secret=$(yq-v4 .clientSecret < "$azure_auth_location")
+  azure_auth_tenant_id=$(yq-v4 .tenantId < "$azure_auth_location")
+  azure_subscription_id=$(yq-v4 .subscriptionId < "$azure_auth_location")
+  az login --service-principal -u "$azure_auth_client_id" -p "$azure_auth_client_secret"\
+    --tenant "$azure_auth_tenant_id" --output none
+  az account set --subscription "${azure_subscription_id}"
+  echo "Setting up the boot image for the ${ADDITIONAL_WORKER_ARCHITECTURE} workers..."
+  vhd_url=$(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -oyaml | \
+    yq-v4 ".data.stream \
+         | eval(.).architectures.${ADDITIONAL_WORKER_ARCHITECTURE}.\"rhel-coreos-extensions\".\"azure-disk\".url")
+  vhd_name=$(basename "${vhd_url}")
+  infra_id=$(yq-v4 '.infraID' < "${SHARED_DIR}"/metadata.json)
+  rg_name="${infra_id}-rg"
+  sa_name=$(az storage account list -g "${rg_name}" | yq-v4 '.[] | select(.name == "cluster*").name')
+  AZURE_STORAGE_KEY=$(az storage account keys list -g "${rg_name}" --account-name "${sa_name}" --query "[0].value" -o tsv)
+  export AZURE_STORAGE_KEY
+  az storage blob copy start --account-name "${sa_name}" \
+      --destination-blob "${vhd_name}" --destination-container vhd --source-uri "$vhd_url"
+  gallery_name=$(az sig list -g "${rg_name}" | yq-v4 '.[].name')
+  image_name="${infra_id}-gen2-${ADDITIONAL_WORKER_ARCHITECTURE}"
```

Comment:
@aleskandro  with the image_name test pass ?

Reply:
yes, it does. You can find the history of the installation jobs with the successful scale-out on Azure starting from the links in the description.